src/sdam/monitor.ts

```diff
@@ -279,11 +279,13 @@ function checkServer(monitor: Monitor, callback: Callback<Document | null>) {
       // if we are using the streaming protocol then we immediately issue another `started`
       // event, otherwise the "check" is complete and return to the main monitor loop
       if (isAwaitable && hello.topologyVersion) {
-        monitor.emit(
-          Server.SERVER_HEARTBEAT_STARTED,
-          new ServerHeartbeatStartedEvent(monitor.address)
-        );
-        start = now();
+        setTimeout(function () {
```

Comment:
can we add a test for the control yield behavior or any test that captures the intended behavior here? the change itself seems straightforward enough; since nothing in our code is listening for this, I don't think this would have any sort of adverse impact on existing monitoring and monitoring-adjacent logic, but I want to make sure, if this is indeed solving the heartbeat issue, that we have some sort of documentation for posterity that states why this is important to have in a timeout; at a minimum I'd like **a comment explaining why** we want to yield control specifically *here* prior to returning, but preferably a regression (unit) test that demonstrates the important flow

if a test is possible but may take a while, I don't mind it being in a follow up PR if we're trying to expedite this PR specifically